src/controllers/authenticationController.ts

```diff
@@ -45,7 +43,8 @@ export default class AuthenticationController {
    */
   public async acs(
     // tslint:disable-next-line:no-any
-    userPayload: unknown
+    userPayload: unknown,
+    res: Response
```

Comment:
this breaks referential transparency and causes side effects - you can actually augment the `IResponsePermanentRedirect` object below with the cookie method:

```ts

function withCookie()

const redirectResponse = ResponsePermanentRedirect({
 href: this.clientSpidAccessRedirectionUrl
});

return {
  ...redirectResponse,
  apply: res => 
};

```